libmscore/textbase.cpp

```diff
@@ -936,11 +936,34 @@ void TextBlock::layout(TextBase* t)
                         break;
                   }
             }
+
       if (_fragments.empty()) {
             QFontMetricsF fm = t->fontMetrics();
             _bbox.setRect(0.0, -fm.ascent(), 1.0, fm.descent());
             _lineSpacing = fm.lineSpacing();
             }
+      else if (_fragments.size() == 1 && _fragments.at(0).text == "") {
```

Comment:
I trust you on this, but, why does this turn out to only cause probems for the leading blank lines, not trailing ones?

Also, is comparing against "" really OK here, what if that initial line actually has spaces?

Reply:
It's comparing against both size of 1 and `""`. If there is only one fragment and it is empty then we have an empty line that has a fragment to save formatting information.
If you have a line of pure spaces it enters the 'else' block which calculates the bounding box based on the text. But since spaces have no bounding box the line "does not exist".

As far as why it does not cause problems for trailing lines -> :man_shrugging:

Reply:
Follow up for completeness: `genText` ignored formatting information for empty lines. That meant that when `createLayout` deleted all fragments and reinserted them it did not have the information provided by empty text fragments. That's why leading fragments did not have the desired effect but trailing ones did (since the information was provided by the line of text whose formatting information was saved by `genText`).